types/storybook-readme/index.d.ts

```diff
@@ -0,0 +1,20 @@
+// Type definitions for storybook-readme 4.0
+// Project: https://github.com/tuchk4/storybook-readme
+// Definitions by: Taeheon Kim  <https://github.com/lonyele>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.9
+
+import { WithReadme } from "./src/withReadme";
+import { WithDocs } from "./src/withDocs";
+import { Doc } from "./src/doc";
+
+export const withDocs: WithDocs;
+export const withReadme: WithReadme;
+export const doc: Doc;
+
+// I'm not sure how to write subfolder's declaration
+// This worked for my local testing environment. It doesn't look right with DefinitelyTyped.
+// I think It would be better If I can just import at the top level such as withReadme, withDocs, or doc.
+// declare module "storybook-readme/components/Marked" {
```

Comment:
As you've apparently discovered already, it's better to give each module its own file. The reason this won't work is that you're already in a module, and `declare module` from within another module tries is a module augmentation, not a new declaration.